src/scripts/settings.ts

```diff
@@ -14,4 +14,13 @@ export module Settings {
 
 		return localResult.Value;
 	}
+
+	export function setSettingsJsonForTesting(jsonObject?: {}): void {
```

Comment:
Also felt a little uncomfortable with this change, hence the naming making it explicit for testing purposes.

I can open a bug to make this DI change you suggested in another change #Pending